### PR TITLE
fix: regex to match require statement without .js extension

### LINKED ISSUE #36283: (Test Fail) Cleanup your project with Modules
**Describe your problem and how to reproduce it:**
To pass the test `  - text: Modules present
    testString: getUserInput => $.get(getUserInput('url')+ '/_api/server.js') .then(data => { assert.match(data, /require.*("|').\/routes.js("|')/gi, 'You should have required your new files'); assert.match(data, /mongo.connect[^]*routes/gi, 'Your new modules should be called after your connection to the database'); }, xhr => { throw new Error(xhr.statusText); })`
The require statement must include the '.js' file extension.

**Add a Link to the page with the problem:**
https://learn.freecodecamp.org/information-security-and-quality-assurance/advanced-node-and-express/clean-up-your-project-with-modules/

To pass this test your require statement must be 
`const routes=require('./routes.js');` instead of `const routes=require('./routes');` which is also valid.

### PR DESCRIPTION
**Pull request by kozlovvski**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #36283

I fixed regexp to match files without `.js` extension.

Also, previous regexp matched strings like `require("./routes.js')` (with mixed quotation marks) or even stuff like `requirefoo"./routes.js'`

Now it should work properly

### CODE REVIEW

**Comment by kozlovvski** [comment]
@RandellDawson sorry, I force-pushed and [previous pull request](https://github.com/freeCodeCamp/freeCodeCamp/pull/36290) has closed itself.

I took your regexp and includes it in English version of the challenge only. Is everything right now?

**Review by Hardik500 — COMMENTED** [review]
Apparently your regex has a bit of problem. Since it also approves require statements which contains new lines. Try limiting the whitespace or new line.

Screenshot_2

**Comment by RandellDawson** [comment]
@Hardik500 The example you wrote above would still be perfectly valid code and would not error out.  You can add as much white space as you want in JavaScript and it would not change anything.  I could write the following also:
```
var express = require

(
  
  'express'


);
```

**Comment by kozlovvski** [comment]
What should I do then?

**Comment by RandellDawson** [comment]
> What should I do then?

@kozlovvski Nothing else for now.  Another mod will review before it can be merged.

**Review by RandellDawson — APPROVED** [review]

**Review by thecodingaviator — DISMISSED** [review]
LGTM

File: curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/clean-up-your-project-with-modules.english.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ Congratulations- you're at the end of this section of Advanced Node and Express
 ```yml
 tests:
   - text: Modules present
-    testString: getUserInput => $.get(getUserInput('url')+ '/_api/server.js') .then(data => { assert.match(data, /require.*("|').\/routes.js("|')/gi, 'You should have required your new files'); assert.match(data, /mongo.connect[^]*routes/gi, 'Your new modules should be called after your connection to the database'); }, xhr => { throw new Error(xhr.statusText); })
+    testString: getUserInput => $.get(getUserInput('url')+ '/_api/server.js') .then(data => { assert.match(data, /require\s*\(('|")\.\/routes(\.js)?\1\)/gi, 'You should have required your new files'); assert.match(data, /mongo.connect[^]*routes/gi, 'Your new modules should be called after your connection to the database'); }, xhr => { throw new Error(xhr.statusText); })
```
Comment (thecodingaviator) [inline curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/clean-up-your-project-with-modules.english.md]:
```suggestion
    testString: getUserInput => $.get(getUserInput('url')+ '/_api/server.js') .then(data => { assert.match(data, /require[^\S\r\n]{1,}*\(('|")\.\/routes(\.js)?\1\)/gi, 'You should have required your new files'); assert.match(data, /mongo.connect[^]*routes/gi, 'Your new modules should be called after your connection to the database'); }, xhr => { throw new Error(xhr.statusText); })
```
@RandellDawson How does this look?

**Review by thecodingaviator — CHANGES_REQUESTED** [review]

**Review by thecodingaviator — APPROVED** [review]
LGTM

**Pull request merged**